subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyCollector.java

```diff
@@ -215,4 +246,11 @@ public interface DependencyCollector {
      */
     Provider<Set<Dependency>> getDependencies();
 
+    /**
+     * Returns all dependency constraints declared on this collector.
```

Comment:
❓ Is this meant to realize the `Provider` constraints?  If so, should note this as an `@implSpec` in the docstring.

Reply:
Not unless you call `.get()` on it, which I think would be obvious? If you think not, I can add it as just normal documentation.